src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+export type AriaLivePoliteness = 'off' | 'polite' | 'assertive';
+
+@Injectable(/*{ providedIn: 'root' }*/)
+export class AriaLiveService implements OnDestroy {
+  private _TICK: number = 100;
+  private _ariaLiveElement: HTMLElement;
+  private _document: Document;
+  private _previousTimeout?: any; // return to `number`, Typescript is missing the correct type for that
+
+  private _id: string =  `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(private _ngZone: NgZone, @Inject(DOCUMENT) _document: any) {
+    this._document = _document;
+    this._ariaLiveElement = this._createLiveElement();
+  }
+
+  private _createLiveElement(): HTMLElement {
+    const ariaLiveElement = this._document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', 'polite');
+
+    this._document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string): Promise<void>;
+  public announce(message: string, politeness?: AriaLivePoliteness): Promise<void>;
+  public announce(message: string, ...args: any[]): Promise<void> {
+    let politeness = 'polite';
+    if (args[0] && typeof args[0] === 'string') {
+      politeness = args[0];
+    }
```

Comment:
This might be able to be cleaned up by using a default param and making `AriaLivePoliteness` an Enum instead of a Type.

```typescript
announce(message = AriaLivePoliteness.off): Promise<void> { }
```